pkg/generators/models/testdata/cases/allof_enum/api.yaml

```diff
@@ -0,0 +1,25 @@
+openapi: 3.0.0
+info:
+  version: 0.1.0
+  title: Test
+
+components:
+  schemas:
+    Artist:
+      type: object
+      properties:
+        leftHand:
+          allOf:
+            - $ref: "#/components/schemas/Color"
+            - nullable: true
+        rightHand:
+          allOf:
+            - $ref: "#/components/schemas/Color"
```

Comment:
This does not test much since the reference the same `Color` type. Types/Enums should be different so we can check that their options are merged.

Reply:
I updated the test to show that it merges it

Reply:
Looks like it's not just the tests, but you also fixed a bug ;)

Reply:
I added support for the very rare usecase of merging two named enums into another one. But technically, yes, I fixed a bug :D